code/Control/UserDefinedFormController.php

```diff
@@ -308,54 +308,79 @@ public function process($data, $form)
 
             if (!empty($data[$field->Name])) {
                 if (in_array(EditableFileField::class, $field->getClassAncestry() ?? [])) {
-                    if (!empty($_FILES[$field->Name]['name'])) {
+                    if (($names = $_FILES[$field->Name]['name'] ?? '') && (!is_array($names) || array_filter($names))) {
                         if (!$field->getFolderExists()) {
                             $field->createProtectedFolder();
                         }
 
-                        $file = Versioned::withVersionedMode(function () use ($field, $form) {
-                            $stage = Injector::inst()->get(UserDefinedFormController::class)->config()->get('file_upload_stage');
-                            Versioned::set_stage($stage);
-
-                            $foldername = $field->getFormField()->getFolderName();
-                            // create the file from post data
-                            $upload = Upload::create();
-                            try {
-                                $upload->loadIntoFile($_FILES[$field->Name], null, $foldername);
-                            } catch (ValidationException $e) {
-                                $validationResult = $e->getResult();
-                                foreach ($validationResult->getMessages() as $message) {
-                                    $form->sessionMessage($message['message'], ValidationResult::TYPE_ERROR);
+                        $processFile = function ($fileData) use ($field, $form, &$attachments) {
```

Comment:
Please move this callback to be a private method in this class instead of an anonymous function.